src/tests/class_tests/openms/source/AASequence_test.cpp

```diff
@@ -1161,6 +1165,42 @@ START_SECTION([EXTRA] testing terminal modifications)
 }
 END_SECTION
 
+START_SECTION([EXTRA] multithreaded example)
+{
+  // All measurements are best of three (wall time, Linux, 8 threads)
+  //
+  // Serial execution of code:
+  // 5e4 iterations -> 5.42 seconds
+  //
+  // Parallel execution of code:
+  // original code:
+  // 5e4 iterations -> 4.01 seconds with boost::shared_mutex
+  // 5e4 iterations -> 8.05 seconds with std::mutex
+  //
+  // refactored code that throws exception outside block:
+  // 5e4 iterations -> 8.55 seconds with omp critical
+  // 5e4 iterations -> 8.59 seconds with std::mutex
+  // 5e4 iterations -> 3.94 seconds with boost::shared_mutex
+  int nr_iterations (5e4), test (0);
+
+#ifdef _OPENMP
+#pragma omp parallel for
+#endif
+  for (int k = 1; k < nr_iterations + 1; k++)
+  {
+    auto aa = AASequence::fromString("TEST[" +  String(0.14*k) + "]PEPTIDE");
+
+#ifdef _OPENMP
+#pragma omp critical (add_test)
+#endif
+    {
+      test += aa.size();
```

Comment:
this should not be done within the loop (since it adds a constant overhead and biases the measurements). use omp-reduction instead.